### PR TITLE
ROX-15587: Add basic Prometheus metrics

### PR DESCRIPTION
**Pull request by BradLugo**
## Description

Implements the following custom Prometheus metrics:

- Counter for flavors used

### CODE REVIEW

**Comment by BradLugo** [comment]
Tested via:
```
❯ nohup kubectl -n infra port-forward svc/infra-server-service 8443:8443 &
[1] 55446
appending output to nohup.out
❯ curl --insecure https://localhost:8443/metrics
... # gRPC and HTTP metrics

# HELP infra_flavors_used Kubernetes cluster flavors being created
# TYPE infra_flavors_used counter
infra_flavors_used{flavor="simulate"} 7
infra_flavors_used{flavor="test-connect-artifact"} 1
infra_flavors_used{flavor="test-hello-world"} 5
infra_flavors_used{flavor="test-qa-demo"} 7

# gRPC and HTTP metrics
...
```

After provisioning a `gke-default` cluster:
```
❯ curl --insecure https://localhost:8443/metrics
... # gRPC and HTTP metrics

# HELP infra_flavors_used Kubernetes cluster flavors being created
# TYPE infra_flavors_used counter
infra_flavors_used{flavor="gke-default"} 1
infra_flavors_used{flavor="simulate"} 7
infra_flavors_used{flavor="test-connect-artifact"} 1
infra_flavors_used{flavor="test-hello-world"} 5
infra_flavors_used{flavor="test-qa-demo"} 7

# gRPC and HTTP metrics
...
```

**Review by tommartensen — CHANGES_REQUESTED** [review]
I'm blocking this because the metrics are publicly accessible without authentication. The metrics themselves look ok.

File: service/cluster/cluster.go [hunk 1]
```diff
@@ -402,6 +404,8 @@ func (s *clusterImpl) create(req *v1.CreateClusterRequest, owner, eventID string
 		"cluster-owner", owner,
 	)
 
+	metrics.FlavorsUsedCounter.WithLabelValues(flav.ID).Inc()
```
Comment (tommartensen) [inline service/cluster/cluster.go:407]: IMHO this should be moved to after the `s.argoWorkflowsClient.CreateWorkflow()` call, where we handle the other post-workflow creation actions, like logging, BigQuery integration.

**Review by tommartensen — CHANGES_REQUESTED** [review]
some thoughts.

**Comment by tommartensen** [comment]
> I'm blocking this because the metrics are publicly accessible without authentication. The metrics themselves look ok.

Addressed in #1222 .

File: service/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	// FlavorsUsedCounter is a Prometheus metric, counting the number of clusters created per flavor
+	FlavorsUsedCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "infra",
+			Name:      "flavors_used",
+			Help:      "Number of cluster created by flavor",
```
Comment (tommartensen) [inline service/metrics/metrics.go:11]:
```suggestion
			Help:      "Number of clusters created by flavor",
```

sorry for not spotting earlier / providing wrong suggestion.

**Review by tommartensen — APPROVED** [review]
LGTM to me with one typo suggestion

**Pull request merged**